src/gettext.cpp

```diff
@@ -234,25 +238,26 @@ void GetText::loadCatalog()
             entryDescriptor.key = &readBuffer.front();
         }
 
-        if(!iconv_cd_ && codepage_ != "UTF-8")
-            iconv_cd_ = iconv_open(this->codepage_.c_str(), "UTF-8");
+        IconvHandle iconv_cd;
+        if(codepage_ != "UTF-8")
+            iconv_cd.reset(iconv_open(this->codepage_.c_str(), "UTF-8"));
 
-        for(auto& entryDescriptor : entryDescriptors)
+        for(const auto& entryDescriptor : entryDescriptors)
         {
             readBuffer.resize(entryDescriptor.valueLen);
 
             file.setPosition(entryDescriptor.valueOffset);
             file.read(&readBuffer.front(), entryDescriptor.valueLen);
 
-            if(iconv_cd_ != nullptr)
+            if(iconv_cd)
             {
                 iconvBuffer.resize(entryDescriptor.valueLen * 6); // UTF needs at most 6 times the size per char, so this should be enough
-                size_t ilength = entryDescriptor.valueLen - 1;    // Don't count terminating zero
-                size_t olength = iconvBuffer.size();
+                size_t iLength = entryDescriptor.valueLen - 1;    // Don't count terminating zero
```

Comment:
why this mixture of camelCase and snake_case? (`iconv_cd`)

Reply:
Because it was like this before ;) As mentioned didn't want to change to much so I was just changing what I needed and a few random things adjacent to it as I've seen them

Reply:
https://github.com/Return-To-The-Roots/mygettext/pull/8/commits/0bbadc80e114eb3653cc80ead2490450ed1be0f2